packages/react-scripts/bin/react-scripts.js

```diff
@@ -8,6 +8,163 @@
+          chalk.red(
+            `There might be a problem with the project dependency tree.\n` +
+              `It is likely ${chalk.bold(
+                'not'
+              )} a bug in Create React App, but something you need to fix locally.\n\n`
+          ) +
+            `The ${chalk.bold(
+              'react-scripts'
+            )} package provided by Create React App requires a dependency:\n\n` +
+            chalk.green(
+              `  "${chalk.bold(dep)}": "${chalk.bold(expectedVersion)}"\n\n`
+            ) +
+            `However, a different version of ${chalk.bold(
+              dep
+            )} was detected higher up in the tree:\n\n` +
+            `  ${chalk.bold(chalk.red(maybeDep))} (version: ${chalk.bold(
+              chalk.red(depPackageJson.version)
+            )}) \n\n` +
+            `This is known to cause hard-to-debug issues.\n` +
+            `Try following these steps:\n\n` +
+            `  ${chalk.cyan('1.')} Delete ${chalk.bold(
+              'package-lock.json'
+            )} (${chalk.underline('not')} ${chalk.bold(
+              'package.json'
+            )}!) and/or ${chalk.bold(
+              'yarn.lock'
+            )} in your project folder.\n\n` +
+            `  ${chalk.cyan('2.')} Delete ${chalk.bold(
+              'node_modules'
+            )} in your project folder.\n\n` +
+            `  ${chalk.cyan('3.')} Remove "${chalk.bold(
+              dep
+            )}" from ${chalk.bold('dependencies')} and/or ${chalk.bold(
+              'devDependencies'
+            )} in the ${chalk.bold(
+              'package.json'
+            )} file in your project folder.\n\n` +
+            `  ${chalk.cyan('4.')} Run ${chalk.bold(
+              'npm install'
+            )} or ${chalk.bold(
+              'yarn'
+            )}, depending on the package manager you use.\n\n` +
+            `If this has not helped, there are a few other things to try:\n\n` +
+            `  ${chalk.cyan('5.')} If you used ${chalk.bold(
+              'npm'
+            )}, install ${chalk.bold(
+              'yarn'
+            )} (http://yarnpkg.com/) and repeat the above steps with it instead.\n` +
+            `     This may help because npm has known issues with package hoisting which may get resolved in future versions.\n\n` +
+            `  ${chalk.cyan('6.')} Check if ${chalk.bold(
+              maybeDep
+            )} is outside your project directory.\n` +
+            `     For example, you might have installed something accidentally in your home folder.\n\n` +
+            `  ${chalk.cyan('7.')} Try running ${chalk.bold(
+              `npm ls ${dep}`
+            )} in your project folder.\n` +
+            `     This will tell you which ${chalk.underline(
+              'other'
+            )} package (apart from the expected ${chalk.bold(
+              'react-scripts'
+            )}) installed ${chalk.bold(dep)}.\n\n` +
+            `If nothing else helps, add ${chalk.bold(
```

Comment:
Should we mention valid cases?

When not using YW, it's perfectly valid to nest like so:
```
/
/node_modules
  /eslint@3.x
/client
  /node_modules
    /react-scripts@2.x
    /eslint@4.x (hoisted)
```

Reply:
Open to other wordings!